web3/contract.py

```diff
@@ -638,6 +660,103 @@ def _encode_constructor_data(cls, args=None, kwargs=None):
         return deploy_data
 
 
+class ContractConstructor:
```

Comment:
Is there enough code overlap between this class and the ContractFunction class, to attempt to unify them in some way? If the main difference is the transaction parameter checking, you could use your strategy of putting the parameter check into a function (`check_forbidden_keys_in_transaction`) in a shared base class that can be overrided or have class or instance level `forbidden_keys` set to fit the different situation.

Reply:
That's a good point. I will try to do that.